providers/core/core.battery.js

```diff
@@ -0,0 +1,78 @@
+/*globals navigator*/
+/*jshint node:true*/
+var PromiseCompat = require('es6-promise').Promise;
+var nav = navigator;
+
+var BatteryProvider = function(cap, dispatchEvent) {
+  "use strict";
+  this._batteryPromise = nav && nav.getBattery ? nav.getBattery() : null;
+
+  if (this._batteryPromise) {
+    this._batteryPromise.then(function(battery) {
+      this._setupListeners(battery, dispatchEvent);
+    }.bind(this));
+  }
+};
+
+BatteryProvider.prototype.getCharging = function() {
+  "use strict";
+  if (!this._batteryPromise) {
+    return PromiseCompat.resolve(true);  // Charging / plugged in
+  }
+  return this._batteryPromise.then(function(battery) {
```

Comment:
The resulting object is dynamic.  (I just checked at the JS console on my laptop.)